message/validation/consensus_validation.go

```diff
@@ -0,0 +1,361 @@
+	"github.com/bloxapp/ssv/protocol/v2/qbft/roundtimer"
+	"github.com/bloxapp/ssv/protocol/v2/ssv/queue"
+	ssvtypes "github.com/bloxapp/ssv/protocol/v2/types"
+)
+
+func (mv *MessageValidator) validateConsensusMessage(share *ssvtypes.SSVShare, msg *queue.DecodedSSVMessage, nonCommittee bool, receivedAt time.Time) error {
+	signedMsg, ok := msg.Body.(*specqbft.SignedMessage)
+	if !ok {
+		return fmt.Errorf("expected consensus message")
+	}
+
+	if len(msg.Data) > maxConsensusMsgSize {
+		return fmt.Errorf("size exceeded")
+	}
+
+	if err := mv.validateSignatureFormat(signedMsg.Signature); err != nil {
+		return err
+	}
+
+	if err := mv.validConsensusSigners(share, signedMsg); err != nil {
+		return err
+	}
+
+	if nonCommittee && (signedMsg.Message.MsgType != specqbft.CommitMsgType || !share.HasQuorum(len(signedMsg.Signers))) {
+		e := ErrNonCommitteeOnlyDecided
+		e.got = fmt.Sprintf("%v (%v signers)", signedMsg.Message.MsgType, len(signedMsg.Signers))
+		return e
+	}
+
+	msgSlot := phase0.Slot(signedMsg.Message.Height)
+	msgRound := signedMsg.Message.Round
+
+	role := msg.GetID().GetRoleType()
+	maxRound := mv.maxRound(role)
+
+	if !nonCommittee {
+		if err := mv.validateSlotTime(msgSlot, role, receivedAt); err != nil {
+			return err
+		}
+
+		if msgRound > maxRound {
+			err := ErrRoundTooHigh
+			err.got = fmt.Sprintf("%v (%v role)", msgRound, role)
+			err.want = fmt.Sprintf("%v (%v role)", maxRound, role)
+			return err
+		}
+	}
+
+	// TODO: think about correct implementation of estimated round checks and then enable it
+	//slotStartTime := mv.netCfg.Beacon.GetSlotStartTime(messageSlot).
+	//	Add(mv.waitAfterSlotStart(role)) // TODO: do we need this?
+	//
+	//sinceSlotStart := time.Duration(0)
+	//estimatedRound := specqbft.FirstRound
+	//if receivedAt.After(slotStartTime) {
+	//	sinceSlotStart = receivedAt.Sub(slotStartTime)
+	//	estimatedRound = mv.currentEstimatedRound(sinceSlotStart)
+	//}
+	//
+	//lowestAllowed := estimatedRound - allowedRoundsInPast
+	//highestAllowed := estimatedRound + allowedRoundsInFuture
+	//
+	//if msgRound < lowestAllowed || msgRound > highestAllowed {
+	//	err := ErrEstimatedRoundTooFar
+	//	err.got = fmt.Sprintf("%v (%v role)", msgRound, role)
+	//	err.want = fmt.Sprintf("between %v and %v (%v role) / %v passed", lowestAllowed, highestAllowed, role, sinceSlotStart)
+	//	return err
+	//}
+
+	if mv.hasFullData(signedMsg) {
+		hashedFullData, err := specqbft.HashDataRoot(signedMsg.FullData)
+		if err != nil {
+			return fmt.Errorf("hash data root: %w", err)
+		}
+
+		if hashedFullData != signedMsg.Message.Root {
+			return ErrInvalidHash
+		}
+	}
+
+	consensusID := ConsensusID{
+		PubKey: phase0.BLSPubKey(msg.GetID().GetPubKey()),
+		Role:   role,
+	}
+	state := mv.consensusState(consensusID)
+
+	if !nonCommittee {
+		for _, signer := range signedMsg.Signers {
+			if err := mv.validateSignerBehavior(state, signer, share, msg); err != nil {
+				return fmt.Errorf("bad signer behavior: %w", err)
+			}
+		}
+	}
```

Comment:
why do we skip this check (and the one below) if the validator is non-committee? did i miss it in the [knowledge-base](https://github.com/bloxapp/knowledge-base)?